docs/examples/adding_events.md

```diff
@@ -0,0 +1,81 @@
+# Adding events to your bot
+
+Player state changes get emitted as events. You can attach to a particular listener to keep track of them.
+
+List of all available events : [Link](https://discord-player.js.org/docs/types/discord-player/guildqueueevent)
+
+The most commonly used listeners and their sample implementation are listed below :
+
+Add these in your _index.js_ where you initialized the Discord Player instance
+
+## Debug events
+
+```js
+player.on('debug', async (message) => {
+    //Emitted when the player sends debug info
+    //Useful for seeing what dependencies, extractors, etc are loaded
+    console.log(`General player debug event: ${message}`);
+});
+
+player.events.on('debug', async (queue, message) => {
+    //Emitted when the player queue sends debug info
+    //Useful for seeing what state the current queue is at
+    console.log(`Player debug event: ${message}`);
+});
+```
+
+## Error events
+
+```js
+player.events.on('error', (queue, error) => {
+    //Emitted when the player queue encounters error
+    console.log(`General player error event: ${error.message}`);
+    console.log(error);
+});
+
+player.events.on('playerError', (queue, error) => {
+    //Emitted when the audio player errors while streaming audio track
+    console.log(`Player error event: ${error.message}`);
+    console.log(error);
+});
+```
+
+## General events
+
+`queue.metadata.send()` is used to send a message to the interaction channel
```

Comment:
`queue.metadata` by default is undefined and it depends upon what user sets the metadata to. For example, if user sets the metadata to `{ hello: 'world' }` then `queue.metadata.send` wont work. This description needs to be updated.